lib/mutant/ast/regexp/transformer/alternative.rb

```diff
@@ -0,0 +1,37 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        # Transformer for Regexp `alternative` nodes
+        #
+        # This transformer is very similar to the generic recursive mapper
+        # except for the fact that the `Regexp::Expression` class for
+        # `alternative` nodes has a unique constructor
+        class Alternative < self
+          register :regexp_sequence_expression
+
+          # Mapper from `Regexp::Expression` to `Parser::AST::Node`
+          ExpressionToAST = Class.new(Recursive::ExpressionToAST)
+
+          # Mapper from `Parser::AST::Node` to `Regexp::Expression`
+          class ASTToExpression < Transformer::ASTToExpression
+            ALTERNATIVE = IceNine.deep_freeze(
+              ::Regexp::Expression::Alternative.new(0, 0, 0)
```

Comment:
Three numbers that are usually set as a result of parsing a regular expression that don't seem to actually matter for unparsing. I can extract constants or alternatively I could just leave a comment

https://github.com/ammar/regexp_parser/blob/master/lib/regexp_parser/expression/sequence.rb#L10